cpp/tests/quantiles/quantiles_test.cu

```diff
@@ -0,0 +1,554 @@
+            q_expect{ 0.5, 1.0, 0.0, 0.5, 0.5, 0.0 },
+            q_expect{ 1.0, 1.0, 1.0, 1.0, 1.0, 1.0 }
+        }
+    };
+}
+
+// unsorted
+
+template<typename T>
+std::enable_if_t<std::is_floating_point<T>::value, test_case<T>>
+unsorted() {
+    return test_case<T> {
+        fixed_width_column_wrapper<T> ({ 6.8, 0.15, 3.4, 4.17, 2.13, 1.11, -1.00, 0.8, 5.7 }),
+        {
+            q_expect{ 0.0, -1.00, -1.00, -1.00, -1.00, -1.00 },
+        }
+    };
+}
+
+template<typename T>
+std::enable_if_t<std::is_integral<T>::value and not cudf::is_boolean<T>(), test_case<T>>
+unsorted() {
+    return test_case<T> {
+        fixed_width_column_wrapper<T> ({ 6, 0, 3, 4, 2, 1, -1, 1, 6 }),
+        {
+            q_expect{ 0.0, -1, -1, -1, -1, -1 }
+        }
+    };
+}
+
+template<typename T>
+std::enable_if_t<cudf::is_boolean<T>(), test_case<T>>
+unsorted() {
+    return test_case<T> {
+        fixed_width_column_wrapper<T> ({ 0, 0, 1, 1, 0, 1, 1, 0, 1 }),
+        {
+            q_expect{ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,}
+        }
+    };
+}
+
+} // namespace testdata
+
+// =============================================================================
+// ----- helper functions ------------------------------------------------------
+
+template<typename T>
+void test(testdata::test_case<T> test_case) {
+    using namespace cudf::experimental;
+
+    cudf::table_view in_table { { test_case.column } };
+
+    for (auto & expected : test_case.expectations) {
+
+        auto actual_higher = quantiles(in_table, expected.quantile, interpolation::HIGHER, { test_case.col_order });
+        expect_scalars_equal(expected.higher, *actual_higher[0]);
+
+        auto actual_lower = quantiles(in_table, expected.quantile, interpolation::LOWER, { test_case.col_order });
+        expect_scalars_equal(expected.lower, *actual_lower[0]);
+
+        auto actual_linear = quantiles(in_table, expected.quantile, interpolation::LINEAR, { test_case.col_order });
+        expect_scalars_equal(expected.linear, *actual_linear[0]);
+
+        auto actual_midpoint = quantiles(in_table, expected.quantile, interpolation::MIDPOINT, { test_case.col_order });
+        expect_scalars_equal(expected.midpoint, *actual_midpoint[0]);
+
+        auto actual_nearest = quantiles(in_table, expected.quantile, interpolation::NEAREST, { test_case.col_order });
+        expect_scalars_equal(expected.nearest, *actual_nearest[0]);
+    }
+}
+
+// =============================================================================
+// ----- tests -----------------------------------------------------------------
+
+template <typename T>
+struct QuantilesTest : public cudf::test::BaseFixture {
+};
+
+using TestTypes = cudf::test::NumericTypes;
+// using TestTypes = cudf::test::Types<int8_t>;
+// using TestTypes = cudf::test::Types<int8_t, int16_t, int32_t, int64_t>;
+// using TestTypes = cudf::test::Types<float, double>;
+// using TestTypes = cudf::test::Types<cudf::experimental::bool8>;
+
```

Comment:
Missing tests of error cases? Expect throw for unsupported types. Expect throw for mismatched table and sort Info sizes. Expect no throw for empty inputs (expect empty outputs). Etc.